libs/langchain-community/src/vectorstores/azure_cosmosdb.ts

```diff
@@ -355,9 +400,21 @@ export class AzureCosmosDBVectorStore extends VectorStore {
     databaseName: string,
     collectionName: string
   ): Promise<void> {
-    await client.connect();
-    this.database = client.db(databaseName);
-    this.collection = this.database.collection(collectionName);
+    this.connectPromise = (async () => {
+      await client.connect();
+      this.database = client.db(databaseName);
+      this.collection = this.database.collection(collectionName);
+    })();
+
+    // Unless skipCreate is set, create the index
+    // This operation is no-op if the index already exists
+    if (!this.indexOptions.skipCreate) {
```

Comment:
This could be breaking to someone who already has a flow setup which creates an index or fails. Instead, let's make this opt-in, document it well and add a note saying in version `0.3.0` (planned for a few months from now, breaking changes will ship w/ this release) it will change to opt-out.

Reply:
I think worth it